spring-cloud-deployer-local/src/main/java/org/springframework/cloud/deployer/spi/local/LocalAppDeployer.java

```diff
@@ -154,6 +154,9 @@ public void undeploy(String id) {
 			}
 			running.remove(id);
 		}
+		else {
+			throw new IllegalStateException(String.format("App %s has not been deployed.", id));
```

Comment:
This message sounds like we were trying to deploy the app, maybe say "App %s is not in a deployed state."